native/src/main/java/io/ballerina/stdlib/crypto/CryptoUtils.java

```diff
@@ -124,7 +125,14 @@ public static Object hmac(String algorithm, byte[] key, byte[] input) {
      */
     public static byte[] hash(String algorithm, byte[] input, Object salt) {
         try {
-            MessageDigest messageDigest = MessageDigest.getInstance(algorithm);
+            MessageDigest messageDigest;
+
+            if (Constants.KECCAK256.equals(algorithm)) {
+                messageDigest = new Keccak.Digest256();
+            } else {
+                messageDigest = MessageDigest.getInstance(algorithm);
+            }
+
```

Comment:
```suggestion
            CryptoUtils.addBCProvider();
            MessageDigest messageDigest = MessageDigest.getInstance(algorithm);
```
Can't we do this instead ?

Reply:
Yes, That'd be a better option.